lib/ex_doc/formatter/html/templates.ex

```diff
@@ -201,6 +195,24 @@ defmodule ExDoc.Formatter.HTML.Templates do
     ~s/{"id":"#{id}","anchor":"#{URI.encode(anchor)}"}/
   end
 
+  defp sidebar_items_json_string(module_node, items) do
```

Comment:
Let's work with iodata instead of keyword list, as it is quite more efficient. Comprehensions should help too. Something like this:

```elixir
    json_attrs =
      for key <- [:id, :title, :title_truncated],
          value = Map.get(module_node, key),
          do: [json_kv(key, inspect(value)), ?,]

    json_attrs = [json_attrs | json_kv(:group, Atom.to_string(module_node.group))]

    json_attrs =
      case items do
        "" -> json_attrs
        items -> [json_attrs, ?, | json_kv(:nodeGroups, "[#{items}]")]
      end

    IO.iodata_to_binary [?{, json_attrs, ?}]
```

Reply:
Oh, I forgot to include it. It is this tiny function: `defp json_kv(key, val), do: [?", Atom.to_string(key), ?", ?:, value]`